src/powerquery-language-services/library/librarySymbolUtils.ts

```diff
@@ -10,45 +10,61 @@ import { Library, LibraryDefinitionUtils } from "../library";
 import { LibrarySymbol, LibrarySymbolFunctionParameter } from "./librarySymbol";
 import { CompletionItemKind } from "../commonTypes";
 
-export type IncompleteLibrary = {
+// Created when non-zero conversion errors occur.
+export interface IncompleteLibrary {
     readonly library: Library.ILibrary;
-    readonly invalidSymbols: ReadonlyArray<LibrarySymbol>;
-};
+    readonly failedLibrarySymbolConversions: ReadonlyArray<FailedLibrarySymbolConversion>;
+}
 
-export type IncompleteLibraryDefinitions = {
+export interface IncompleteLibraryDefinitions {
     readonly libraryDefinitions: ReadonlyMap<string, Library.TLibraryDefinition>;
-    readonly invalidSymbols: ReadonlyArray<LibrarySymbol>;
-};
+    readonly failedLibrarySymbolConversions: ReadonlyArray<FailedLibrarySymbolConversion>;
+}
+
+export interface FailedLibrarySymbolConversion {
+    readonly librarySymbol: LibrarySymbol;
+    readonly kind: FailedLibrarySymbolConversionKind;
+}
 
+export interface FailedLibrarySymbolParameterConversion extends FailedLibrarySymbolConversion {
+    readonly kind: FailedLibrarySymbolConversionKind.Parameter;
+    readonly index: number;
+}
+
+export enum FailedLibrarySymbolConversionKind {
+    AsPowerQueryType = "AsPowerQueryType",
+    CompletionItemKind = "CompletionItemKind",
+    Parameter = "Parameter",
+    PrimitiveType = "PrimitiveType",
+}
+
+// Always returns a library, even if there are failed library symbol conversions.
```

Comment:
I would add this either as doc comments, or /** */ so that intellisense picks it up